### DISCUSSION TITLE
Tabulated fluid properties in Modelica 4.0.0

### DISCUSSION THREAD

**Discussion by Jake-HW-Miles** [OpenModelica/OpenModelica#12966] [Q&A]
Hi all, I am relatively new to OpenModelica so excuse any ignorance here! I really like the tool and am currently using it to model a nuclear power system, from reactor to steam generator.

I am wanting to use a custom fluid for the reactor working fluid, however am having issues. For now, I have settled with constant properties - however do not want to miss out on phenomena due to such a simplification.

However, when I try something like tabulated fluid properties, I receive the following error:
`The arguments u_min and u_max provided in the function call solveOneNonlinearEquation(f,u_min,u_max) do not bracket the root of the single non-linear equation 0=f(u): u_min = 800 u_max = 995 fa = f(u_min) = 488252 fb = f(u_max) = 614710 fa and fb must have opposite sign which is not the case`

Is there anything else I can do here? The fluid must following the property functions, however I cannot get passed this error.

Alternatively, is there a different method I can use? Function properties?

**Comment by casella** [comment]
Can you share a simple test that reproduces the issue?

**Reply by Jake-HW-Miles** [reply]
Not sure if I should have made any modifications to the file for you to replicate! I have been toying with this for a while so have attached a generic ternary molten salt in **tabulated_salt.txt** and an associated test case ( **fluid_testing_model.txt** ).

The error is analogous to the above, bar the values!
[fluid_testing_model.txt](https://github.com/user-attachments/files/17231627/fluid_testing_model.txt)
[tabulated_salt.txt](https://github.com/user-attachments/files/17231628/tabulated_salt.txt)

**Reply by Jake-HW-Miles** [reply]
Hi @casella, did you have a chance to try and recreate the error I am experiencing?
